resolwe/storage/connectors/transfer.py

```diff
@@ -39,189 +42,32 @@ def retry_on_transfer_error(wrapped, instance, args, kwargs):
-
-    def write(self, data: bytes) -> int:
-        """Write data into the stream.
-
-        :param data: data to be writen to the stream.
-        :type data: bytes
-
-        :return: the number of bytes actually writen.
-        :rtype: int
-        """
-        datamv = memoryview(data)
-        data_pos = 0
-        with self.__buffer_modify:
-            # Wait for enough bytes to become available.
-            # Abort if stream was closed.
-            while data_pos < len(datamv) and not self.closed:
-                while not self.__can_write and not self.closed:
-                    self.__buffer_modify.wait()
-                bytes_to_write = min(self.__bytes_free(), len(datamv) - data_pos)
-                slice_start = self.__head
-                slice_end = slice_start + bytes_to_write
-                if slice_end <= self.__bs:
-                    self.__buffer[slice_start : slice_start + bytes_to_write] = datamv[
-                        data_pos : data_pos + bytes_to_write
-                    ]
-                else:  # rotation
-                    slice_end %= self.__bs
-                    sep = self.__bs - slice_start
-                    self.__buffer[slice_start : self.__bs] = datamv[
-                        data_pos : data_pos + sep
-                    ]
-                    remaining_bytes = bytes_to_write - sep
-                    self.__buffer[0:remaining_bytes] = datamv[
-                        data_pos + sep : data_pos + bytes_to_write
-                    ]
-                self.__head += bytes_to_write
-                self.__head %= self.__bs
-                data_pos += bytes_to_write
-                if self.__reading_bytes > 0:
-                    if self.__bytes_available() >= self.__reading_bytes:
-                        self.__buffer_modify.notify()
-        return data_pos
-
-    def close(self):
-        """Close stream."""
-        self.__closed = True
-        with self.__buffer_modify:
-            self.__buffer_modify.notify()
-
-    def tell(self) -> int:
-        """Get the number of bytes read from the stream.
-
-        :return: number of bytes read from the stream.
-        :rtype: int
-        """
-        return self._bytes_read
-
-
 class Transfer:
     """Transfer data between two storage connectors using in-memory buffer."""
 
     def __init__(
-        self, from_connector: BaseStorageConnector, to_connector: BaseStorageConnector,
+        self,
+        from_connector: "BaseStorageConnector",
+        to_connector: "BaseStorageConnector",
     ):
         """Initialize transfer object."""
         self.from_connector = from_connector
         self.to_connector = to_connector
 
-    def transfer_rec(self, url: str, objects: List[str] = None):
+    def pre_processing(
+        self, url: str, objects: Optional[List[str]] = None
+    ) -> List[str]:
+        """Notify connectors that transfer is about to start."""
+        objects = self.from_connector.before_get(objects, url)
```

Comment:
Why is this the only function that can modify objects?

Reply:
Actually objects is currently not really used anywhere in pre and post-processing so I will remove it.

Reply:
I was too fast. It is used to clean after backup is transfered to another location.

However the modification of objects is currently not needed so I removed it.